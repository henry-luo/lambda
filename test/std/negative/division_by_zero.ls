// Test: Division by Zero
// Category: negative
// Type: negative
// Expected: error

// This test should cause a division by zero
let a = 10
let b = 0
a / b
