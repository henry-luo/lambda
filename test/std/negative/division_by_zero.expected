09:15:53 [NOTICE] Start transpiling test/std/negative/division_by_zero.ls...
[script:list]
  [content:list[3]]
    no declare
    item:
    [let stam:null]
      declare:
      [assign expr:a:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [let stam:null]
      declare:
      [assign expr:b:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [binary expr /.7:float]
      [primary expr:int,const:1]
        [ident:a:int,const:1]
      [primary expr:int,const:1]
        [ident:b:int,const:1]
09:15:53 [NOTICE] compiling C code in 'test/std/negative/division_by_zero.ls' to MIR
C2MIR init end           -- 71 usec
test/std/negative/division_by_zero.ls:1:2: warning -- unknown pragma
    preprocessor tokens -- 2365, parse tokens -- 1415
  C2MIR preprocessor end    -- 503 usec
  C2MIR parser end          -- 622 usec
  C2MIR context checker end -- 691 usec
  C2MIR generator end       -- 731 usec
C2MIR compiler end                -- 760 usec
09:15:53 [NOTICE] Generating native code...
##### Script 'test/std/negative/division_by_zero.ls' executed: #####
inf
